### PR TITLE
Add debug logs to help troubleshoot "--wait" flag

### PR DESCRIPTION
**Pull request by pchico83**
Signed-off-by: Pablo Chico de Guzman <pchico83@gmail.com>

Add debug logs to the `--wait` flag to help troubleshoot https://github.com/okteto/app/issues/4112

### CODE REVIEW

File: cmd/pipeline/deploy.go [hunk 1]
```diff
@@ -285,7 +286,8 @@ func waitForResourcesToBeRunning(ctx context.Context, name string, timeout time.
 				return err
 			}
 			allRunning := true
-			for _, status := range resourceStatus {
+			for resourceID, status := range resourceStatus {
```
Comment (rberrelleza) [inline cmd/pipeline/deploy.go:289]: What is the value of resource ID?

Comment (rberrelleza) [reply cmd/pipeline/deploy.go:289]: Can we use type/name instead? The resourceID is not human readable, and we don't expose it anywhere in the UI.

File: pkg/okteto/pipeline.go [hunk 2]
```diff
@@ -404,18 +407,21 @@ func (c *OktetoClient) GetResourcesStatusFromPipeline(ctx context.Context, name
 	status := make(map[string]string)
 	for _, d := range queryStruct.Space.Deployments {
 		if string(d.DeployedBy) == name {
-			status[string(d.ID)] = string(d.Status)
+			resourceName := fmt.Sprintf("deployment/%s", d.Name)
```
Comment (jmacelroy) [inline pkg/okteto/pipeline.go:410]: The logic for the key creation to access an element in the status map is duplicated and prone to bugs. Can we consolidate it into a function to eliminate the duplication and reduce the change of a bug occurring by an update to the key creation in either status write or read and not in the other location? Probably, something that takes a map and some constant indicating the resource type and either applies the change or returns what should be used for the key.

Comment (jmacelroy) [reply pkg/okteto/pipeline.go:410]: A test may make sense for that function as well as it should be easy and help ensure that we have the correct debug code we want in place when the opportunity to capitalize on the debug logs occurs.

**Review by ifbyol — APPROVED** [review]

File: cmd/pipeline/deploy.go [hunk 3]
```diff
@@ -26,6 +26,7 @@ import (
 	"github.com/okteto/okteto/pkg/cmd/pipeline"
 	oktetoErrors "github.com/okteto/okteto/pkg/errors"
 	"github.com/okteto/okteto/pkg/k8s/configmaps"
+	"github.com/okteto/okteto/pkg/log"
```
Comment (jLopezbarb) [inline cmd/pipeline/deploy.go:29]: already declared with oktetoLog on the next line

**Review by jLopezbarb — APPROVED** [review]

**Pull request merged**